Add option for generated enum types

gslaszlo opened the pull request:
Add support for generated enum types, as in [orval docs](https://orval.dev/reference/configuration/output#enumgenerationtype).

gslaszlo committed:
chore: update openapi-typescript to version 7.8.0 and orval to version 7.10.0

gslaszlo committed:
feat: add enum generation type support

gslaszlo committed:
docs: update README to include new enum generation options

gslaszlo committed:
docs: fix markdown

gslaszlo committed:
docs: correct validation function documentation for enum generation type

DefCon-007 commented:
@gslaszlo Thanks a lot for the changes. The changes look good. I will try to test it locally once, meanwhile, can you add some unit tests for this? We can add some functional tests like this,  https://github.com/grafana/openapi-to-k6/blob/main/tests/functional-tests/test-generator/generator.test.ts,  we could take a single schema and test it out for all the supported enum types.

gslaszlo committed:
Add tests for enumGenerationType

gslaszlo commented:
Thanks for pointing it out, I've added tests for all 3 cases. I didn't want to do a lot of edits in the `generator.test.ts` so I added 3 fixtures with different `expected_sdk` params, but if you prefer to add more code instead of fixtures, I can do that as well.

DefCon-007 commented:
@gslaszlo Could you fix the conflicts? We are good to merge once they are fixed.

gslaszlo committed:
Merge branch 'main' into main

gslaszlo commented:
@DefCon-007 Sorry for the delay, fixed the conflicts.
